File: SpeedTestGameClient/core/src/main/java/si/um/feri/speedii/dao/MeasurementCRUD.java
package si.um.feri.speedii.dao;

import java.io.IOException;
import java.time.LocalDateTime;
import java.util.List;

import si.um.feri.speedii.classes.Measurement;
import si.um.feri.speedii.classes.User;

public interface MeasurementCRUD extends DaoCrud<Measurement> {

    List<Measurement> getByUser(User user) throws IOException;

    List<Measurement> getByTimeFrame(LocalDateTime start, LocalDateTime end) throws IOException;

    boolean insertMany(List<Measurement> list) throws IOException;
}

File: SpeedTestClient/src/main/kotlin/dao/MeasurementCrud.kt
package dao

import Measurment
import User
import java.time.LocalDateTime

interface MeasurementCrud : DaoCrud<Measurment> {
    suspend fun getByUser(user: User): List<Measurment>
    suspend fun getByTimeFrame(start: LocalDateTime, end: LocalDateTime): List<Measurment>
}